### DISCUSSION TITLE
Running in Alpine Docker Container

### DISCUSSION THREAD

**Discussion by jarrodsfarrell** [diesel-rs/diesel#3014] [Q&A]
Omitting some information given this is a company project, but I've been trying to build a program within Alpine however running something akin to:
```rs
let connection = MysqlConnection::establish(&data_url)?;
```
kills the program immediately without a unwrap or panic or anything useful; just a blank line.

I'ved tried a couple different packages I could think of but all failed. This is my Dockerfile:
```dockerfile
FROM rust:1.57-alpine3.14 as builder
RUN apk add --no-cache mariadb-connector-c-dev musl-dev
WORKDIR /usr/src/CENSORED
COPY . .
RUN cargo install --debug --path .

FROM alpine:3.14
RUN apk add --no-cache libgcc mariadb-connector-c mariadb-client mariadb-common
COPY --from=builder /usr/local/cargo/bin/CENSORED /usr/local/bin/CENSORED
ENTRYPOINT ["/usr/local/bin/CENSORED"]
CMD ["--help"]
```

---
# MRE
```rs
use diesel::{MysqlConnection, Connection};

fn main() {
    println!("Trying to establish a connection to a nonexistent database...");
    MysqlConnection::establish("mysql://nonexistent/database").err();
    println!("We didn't crash!");
}
```

```toml
[package]
name = "alpine-diesel-mre"
version = "0.1.0"
edition = "2021"

[dependencies]

[dependencies.diesel]
version = "1.4.8"
features = ["mysql"]
```

```dockerfile
FROM rust:1.57-alpine3.14 as builder
RUN apk add --no-cache mariadb-connector-c-dev musl-dev
WORKDIR /usr/src/alpine-diesel-mre
COPY . .
RUN cargo install --debug --path .


FROM alpine:3.14
RUN apk add --no-cache mariadb-connector-c musl
COPY --from=builder /usr/local/cargo/bin/alpine-diesel-mre /usr/local/bin/alpine-diesel-mre
ENTRYPOINT ["/usr/local/bin/alpine-diesel-mre"]
```

**Comment by weiznich** [comment]
We cannot answer question that did not contain the required information to reproduce your problem. I see two options here:
* Either you are able to provide at least some code which helps to reproduce the problem. That does not need to be your company code base, just a minimal example will be fine.
* Or you/your company decides that you need help with this, but cannot afford to extract a minimal example/share the code publicly. In this case I can offer paid consultant work.

**Reply by jarrodsfarrell** [reply]
Wrote up a MRE in the OP.

**Reply by weiznich** [reply]
Thanks for the MRE. That sounds like https://github.com/rust-lang/rust/issues/82193 for me. 
The short version of this is that there is an incompatibility between how rustc links the musl-libc and alpine links musl-libc to shared binaries. You can likely fix that by tweaking the flags passed to rustc. See the linked issue for details.
